File: 1863-sum-of-all-subset-xor-totals/1863-sum-of-all-subset-xor-totals.cpp
class Solution {
public:
    int subsetXORSum(vector<int>& nums) {
        int sum = 0, xr = 0;
        recurse(nums, 0, nums.size(), xr, sum);
        return sum;
    }
    void recurse(vector<int>& nums, int i, int n, int& xr, int& sum) {
        if (i == n) {
            sum += xr;
            return;
        }
        xr ^= nums[i];
        recurse(nums, i+1, n, xr, sum);
        xr ^= nums[i];
        recurse(nums, i+1, n, xr, sum);
    }
};

File: 1863-sum-of-all-subset-xor-totals/1863-sum-of-all-subset-xor-totals.py
class Solution:
    def subsetXORSum(self, nums: List[int]) -> int:
        xr, sum = 0, 0
        xr, sum = self.recurse(nums, 0, len(nums), xr, sum)
        return sum
    def recurse(self, nums, i, n, xr, sum) :
        if i == n:
            sum += xr
            return xr, sum
        xr ^= nums[i]
        xr, sum = self.recurse(nums, i+1, n, xr, sum)
        xr ^= nums[i]
        xr, sum = self.recurse(nums, i+1, n, xr, sum)
        return xr, sum